### ISSUE TITLE
prod cluster OpenShift console needs to be externally accessible

### ISSUE THREAD

**Issue by jtriley** [OCP-on-NERC/operations#67]
Users need to be able to access the OpenShift console externally. If we apply the external route label to the existing openshift-console route that would give us:

https://console-openshift-console.apps.shift.nerc.mghpcc.org

We might want to consider adding a custom route to the console to shorten this to just `https://shift.nerc.mghpcc.org`

**Comment by larsks** [comment]
> Users need to be able to access the OpenShift console externally.

I was under the impression that all authenticated cluster access -- including the console -- was going to require VPN. If that's not the case, do we also need to expose the API externally?

**Comment by larsks** [comment]
[This OKD issue](https://github.com/okd-project/okd/discussions/901) suggests that there isn't a great solution.

**Comment by larsks** [comment]
The official docs are changing the console routes are [here](https://docs.openshift.com/container-platform/4.10/web_console/customizing-the-web-console.html#customizing-the-web-console-url_customizing-web-console), but I don't think that applies to our situation.

**Comment by jtriley** [comment]
From what I can gather it seems a rebuild of the prod cluster with shift.nerc.mghpcc.org and the external ingress and API addresses might be the only way forward? That might require some refactoring at several layers given that a tagged interface would be required. Might be possible via kernel params to the discovery ISO but I recall having issues getting that working properly in` the past (e.g. custom network config params getting dropped in the installed OS which lead to VLAN tag info getting lost during the install process causing networking to fail)

**Comment by larsks** [comment]
> From what I can gather it seems a rebuild of the prod cluster with shift.nerc.mghpcc.org and the external ingress and API addresses might be the only way forward?

I think we can actually change the hostname for both the console and the oauth server by following these docs:

- https://docs.openshift.com/container-platform/4.11/web_console/customizing-the-web-console.html#customizing-the-web-console-url_customizing-web-console
- https://docs.openshift.com/container-platform/4.11/authentication/configuring-internal-oauth.html#customizing-the-oauth-server-url_configuring-internal-oauth

> That might require some refactoring at several layers given that a tagged interface would be required.

I'm not sure that's true; we're really only talking about changing names here, rather than changing networks or addresses. We already have the public interface set up.

We might encounter issues bootstrapping the cluster (if the console and oauth hostnames point to an address that won't be available until we configure networking), but that can be solved if we're able to provide split DNS for the shift.nerc.mghpcc.org domain (so that requests from VPN-connected hosts see VPN-reachable addresses, and requests from elsewhere see the public addresses).

But before we consider a rebuild I'd like to try following the docs to which I've linked.

**Comment by joachimweyl** [comment]
@larsks I believe this is resolved. Is there any more work we need to do for this?
